### PR TITLE
cli: don't filter on program by default in merge-logs

### PR DESCRIPTION
**Pull request by tbg**
This behavior can be useful, but more often than not, the glob you're
passing in already matches exactly the right files. If the cockroach
binary wasn't called `cockroach` (this does happen), the previous
default wouldn't match any log files and gave a confusing UX.

Instead, accept any program by default. In cases where a filter is
adequate, one can be specified.

Release note: None

### CODE REVIEW

**Review by ajwerner — APPROVED** [review]
Fine by me. It may have minor conflicts with https://github.com/cockroachdb/cockroach/pull/33067 which attempts to deal with some of your other concerns. Also, I can update the above PR to return an error if no files match.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33319#-:-LUGEXoS0Mi1Uk_7HeAA:b1i4mtx)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by tbg** [comment]
TFTR. I'll hold off on this until you've merged your other work. I do think
that the use case in which we're only looking at CRDB logs is the most
frequent one, so returning an error (while generally a good idea) wouldn't
solve the UX problem. The other default I was considering was
`^cockroach.*$`.

**Review by ajwerner — COMMENTED** [review]
I support `^cockroach.*$`. I'll merge the other change now

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33319#-:-LUGJrkD4xhimx6BzWN3:b70gx5n)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by tbg** [comment]
Changed to `^cockroach.*$`. TFTR!

bors r=ajwerner

**Pull request merged**
